project/src/test/java/com/google/sps/EventSpannerTasksTest.java

```diff
@@ -169,14 +169,41 @@ public void testEventCreationDoGet() throws Exception {
     Assert.assertEquals(returnedEvent.getHost().getEmail(), EMAIL);
   }
 
-  /**
-   * Verify getting event from invalid id in /create-event doGet()
-   */
+  /** Verify getting event from invalid id in /create-event doGet() */
   @Test
   public void testEventCreationDoGetInvalid() throws Exception {
     request.addParameter("eventId", /*invalid id= */ "1");
     new EventCreationServlet().doGet(request, response);
 
     Assert.assertEquals(response.getStatus(), HttpServletResponse.SC_NOT_FOUND);
   }
+
+  /**
+   * Verify registering for event that already has attendee
+   */
+  @Test
+  public void testEventRegistrationDoGet() throws Exception {
+    //Existing event to sign up for
+    SpannerTasks.insertOrUpdateUser(ATTENDEE);
+    Event event =
+        new Event.Builder(EVENT_NAME, DESCRIPTION, LABELS, LOCATION, DATE, TIME, HOST)
+            .setAttendees(new HashSet<User>(Arrays.asList(ATTENDEE)))
+            .build();
+    request.addParameter("eventId", event.getId());   
+    SpannerTasks.insertorUpdateEvent(event);
+    // User that is registering
+    authenticationHelper.setEnvIsLoggedIn(true).setEnvEmail(EMAIL).setEnvAuthDomain("example.com");
+
+    new EventRegistrationServlet().doPost(request, response);
+    Event returnedEvent = new Gson().fromJson(response.getContentAsString(), Event.class);
+
+    Assert.assertEquals(returnedEvent.getName(), EVENT_NAME);
```

Comment:
Can you assert returnedEvent equals to event?

Reply:
No, discussion here https://screenshot.googleplex.com/TuV2Xkwwukf.png. The problem is that the db always returns a 'new' Event obj, so assert equals sees them as not equal